crates/exex/src/atlantic/endpoints.rs

```diff
@@ -0,0 +1,266 @@
+use super::error::SharpSdkError;
+use url::Url;
+
+/// The L2 endpoints.
+#[derive(Debug, Clone)]
+pub struct L2Endpoints {
+    /// Submit an Atlantic proof generation query
+    pub proof_generation: Url,
+    /// Submit an Atlantic trace generation query
+    pub trace_generation: Url,
+    /// Submit an Atlantic trace generation and proof generation query
+    pub trace_generation_proof_generation: Url,
+    /// Submit a L2 Atlantic query
+    pub atlantic_query: Url,
+    /// Submit a L2 Atlantic proof generation and proof verification query
+    pub proof_generation_verification: Url,
+    /// Submit a L2 Atlantic proof verification query
+    pub proof_verification: Url,
+}
```

Comment:
this should probably be an enum

Reply:
I would keep it as a structure because when we initialize the Sharp SDK, we initialize all the endpoints at once so that we can then call them as desired in different functions. 

If we made it an enum, when we do `SharpSdk::new`, we would be forced to specify the L2 endpoint which is not very practical. Here especially since it is only URLs so it is not very heavy.

Reply:
I see your approach but adding a new strum deps or doing some macro for that is really overkilled for now in my opinion. Here we have something clear and readable so that I know that in my Sharp sdk I've all my urls and I can query them as I want. We have less than 10 urls in total, this should be fine to have them like that. The debate would be different with 100 I think.

Reply:
it's just that a struct is not meant to store a set of possible values

Reply:
Final conclusion? you want me to use https://github.com/kkrt-labs/keth/pull/100#pullrequestreview-2438192033?

Reply:
I'm open to other suggestions, including not defining an enum and inlining directly the `str` in each function

Reply:
We could also leverage the trait names to do something like

```
use function_name::named;

impl SharpSdk {
    #[named]
    pub async fn proof_generation(
        &self,
        pie_file: Vec<u8>,
        layout: &str,
        prover: ProverVersion,
    ) -> Result<QueryResponse, SharpSdkError> {
        let endpoint = function_name!().to_string();  // Will return "proof_generation"
        // ... rest of the code
    }
}
```

Reply:
fixed